assembly/assembly-ide-war/src/main/webapp/_app/IDE.html

```diff
@@ -54,12 +54,278 @@
+            };
+
+            /*
+             * Load content of IDE JavaScript
+             */
+            this.loadIDEJavaScript = function() {
+                var msg = "Unable to load IDE JavaScript";
+
+                try {
+                    var request = new XMLHttpRequest();
+
+                    request.onprogress = function(event) {
+                        var percentComplete = event.loaded / Loader.ideJavaScriptFileSize * 100;
+                        var rounded = Math.round(percentComplete);
+                        if (rounded > 100) {
+                            rounded = 100;
+                        }
+
+                        document.getElementById("ide-loader-progress-bar").style.width = rounded + "%";
+                    };
+
+                    request.onload = function() {
+                        if (request.status == 200) {
+                            Loader.injectIDEJavaScript();
+                        } else {
+                            console.error(msg);
+                        }
+                    };
+
+                    request.onerror = request.abort = function () {
+                        console.error(msg);
+                    };
+
+                    request.open("GET", "/_app/" + Loader.ideJavaScriptFileName, true);
+                    request.send();
+                } catch (e) {
+                    console.error(msg, e);
+                }
+            };
+
+            /*
+             * Injects IDE JavaScript
+             */
+            this.injectIDEJavaScript = function() {
+                document.getElementById("ide-loader-progress-bar").style.width = "100%";
+
+                var script = document.createElement("script");
+                script.type = "text/javascript";
+                script.language = "javascript";
+                script.async = true;
+                script.src = "/_app/_app.nocache.js";
+                document.getElementsByTagName("head")[0].appendChild(script);
+            };
+
+        };
+
+        setTimeout(function() {
+            Loader.startLoading();
+            window.parent.postMessage("show-ide", "*");
+        }, 1);
+
     </script>
 
-    <script type="text/javascript" language="javascript" src="/_app/browserNotSupported.js"></script>
-    <script type="text/javascript" language="javascript" async="true" src="/_app/_app.nocache.js"></script>
 </head>
 
-<body style="background-color: #21252b; transition: background-color 0.5s ease;" />
+<body style="background-color: #21252b; transition: background-color 0.5s ease;">
+<div id="ide-loader" style="position: fixed; width: 300px; height: 45px; left: 50%; top: 30%;
+            margin-left: -150px; opacity: 0; transition: all 0.2s ease-in-out;">
+    <div style="position: absolute; left: 0px; top: 0px; width: 300px; height: 30px;
+          font-family: sans-serif; font-size: 14px;
+          line-height: 30px; text-align: center; color: #a0a9b7;">Loading IDE</div>
```

Comment:
is it IDE or the product name ?

Reply:
IDE

Reply:
And the fie is moved to `che-ide-gwt-app/src/main/resources/org/eclipse/che/ide/public/IDE.html`

Reply:
We should probably show the product's icon and display the progress bar below.

Reply:
@vitaliy-guliy 
For the main use case: user will go to the IDE from the UD. In this case, we should preload the IDE's javascript. This way it should be possible to remove this loader, wdyt?

For other use cases (when the dashboard is not present or by-pass) there will be a need for this loader.

Reply:
AFAIK UD is already preloading the javascript files of the IDE

Reply:
We already discussed on this.
When opening IDE on slow network the javascript may be not be preloaded. In this case we have to show the user some sort of progress bar.

Reply:
And even id IDE javascript is preloaded, this loader entertains the user while IDE is initializing.